app/assets/src/components/ui/controls/dropdowns/common/DropdownTrigger.jsx

```diff
@@ -1,9 +1,53 @@
 import React from "react";
 import cx from "classnames";
+import { get } from "lodash/fp";
+
 import PropTypes from "prop-types";
 import cs from "./dropdown_trigger.scss";
 
+// Set of component names that render labels that contain badge counts
+const DROPDOWN_LABEL_OWNERS_WITH_BADGE_COUNTS = new Set([
+  "ThresholdFilterDropdown",
+  "MultipleNestedDropdown",
+]);
+
 class DropdownTrigger extends React.Component {
+  constructor(props) {
+    super(props);
+
+    this.resizeObserver = null;
+    this.labelContainerRef = null;
+    this.labelRef = null;
+
+    this.state = {
+      hideDropdownLabel: false,
+    };
+  }
+
+  componentDidMount() {
+    if (this.labelContainerRef) {
+      this.resizeObserver = new ResizeObserver(this.handleFilterResize);
+      this.resizeObserver.observe(this.labelContainerRef.current);
+    }
+  }
+
+  componentWillUnmount() {
+    if (this.resizeObserver) {
+      this.resizeObserver.disconnect();
+    }
+  }
+
+  handleFilterResize = labelContainers => {
+    const labelContainerWidth = labelContainers[0].contentRect.width - 3.7; // -3.7 removes padding
+    const labelTextWidth = this.labelRef.current.offsetWidth + 1;
+    const badgeCountWidth = 24;
```

Comment:
I would add a comment `badge count width set by semantic-ui` so people understand where this value is ultimately coming from.